parser-typechecker/src/Unison/Syntax/TermParser.hs

```diff
@@ -1014,20 +1020,20 @@ binding = label "binding" $ do
   case typ of
     Nothing -> do
       -- we haven't seen a type annotation, so lookahead to '=' before commit
-      (loc, name, args) <- P.try (lhs <* P.lookAhead (openBlockWith "="))
+      (lhsLoc, name, args) <- P.try (lhs <* P.lookAhead (openBlockWith "="))
       body <- block "="
       verifyRelativeName' (fmap Name.unsafeFromVar name)
-      pure $ mkBinding loc (L.payload name) args body
+      pure $ mkBinding (lhsLoc <> ann body) (L.payload name) args body
     Just (nameT, typ) -> do
-      (_, name, args) <- lhs
+      (lhsLoc, name, args) <- lhs
       verifyRelativeName' (fmap Name.unsafeFromVar name)
       when (L.payload name /= L.payload nameT) $
         customFailure $ SignatureNeedsAccompanyingBody nameT
       body <- block "="
       pure $
         fmap
           (\e -> Term.ann (ann nameT <> ann e) e typ)
-          (mkBinding (ann nameT) (L.payload name) args body)
+          (mkBinding (ann lhsLoc <> ann body) (L.payload name) args body)
```

Comment:
Don't include the type annotation in the annotation of the binding, that's what the type annotation node is for.